hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -92,13 +101,42 @@ public void associateWith(List<Token> tokens, int maxAllowed) {
 		associatedTokens.addAllIds(uniqueIds);
 	}
 
-	public void dissociateWith(List<Token> tokens) {
+	public void dissociateWith(List<Pair<TokenRelationship, TokenRelationship>> relations, final OptionValidator validator) {
 		final Set<Id> uniqueIds = new HashSet<>();
-		for (var token : tokens) {
+		for (var rel : relations) {
+			/*	extraction	 */
+			final var accountRel = rel.getKey();
+			final var treasuryRel = rel.getValue();
+			final var account = accountRel.getAccount();
+			final var treasury = treasuryRel.getAccount();
+			final var token = treasuryRel.getToken();
+			/*	validation	 */
+			validateTrue(!token.isDeleted(), TOKEN_WAS_DELETED, () ->
+					"Given token was deleted");
+			validateTrue(!treasury.getId().equals(account.getId()), ACCOUNT_IS_TREASURY, () ->
```

Comment:
If token is deleted, the treasury must be able to do `TokenDisassociate`. In legacy implementation -> the check is:
```Java
/* Once a token is deleted, this always returns false. */
				if (isTreasuryForToken(aId, tId)) {
					return ACCOUNT_IS_TREASURY;
				}
```
you can see the comment that `isTreasuryForToken` returns false and the treasury can do `TokenDisassociate`